src/Downloader.ts

```diff
@@ -184,37 +184,56 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
+  public setBaseUrls(baseUrl: string, baseUrlForMainPage: string = this.mw.veApiUrl.href){
```

Comment:
This function should take no argument. Values should be set based on `this.mw` and `this.mwCapabilities`.